test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/SelectExpandFunctionalTests.cs

```diff
@@ -1667,6 +1667,96 @@ public void ExpandWithNavigationPropCountWithSearchOptionWorks()
             Assert.NotNull(expandedCountSelectItem.SearchOption);
         }
 
+        // $expand=navProp/fully.qualified.type/$count

```

Comment:
Please add a (negative) test case for fully.qualified.undefinedType and verify meaningful exception (i.e., saying that the type fully.qualified.undefinedType was not found).

Reply:
Done